presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -197,6 +258,27 @@ public static Block collapseIpPrefixes(@SqlType("array(IPPREFIX)") Block unsorte
         return blockBuilder.build();
     }
 
+    @Description("Returns whether ipAddress is a private or reserved IP address that is not globally reachable.")
+    @ScalarFunction("is_private_ip")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean isPrivateIpAddress(@SqlType(StandardTypes.IPADDRESS) Slice ipAddress)
+    {
+        BigInteger ipAsBigInt = toBigInteger(ipAddress);
+        boolean isIPv4 = isIpv4(ipAddress);
+
+        List<BigInteger[]> rangesToCheck = isIPv4 ? privateIPv4AddressRanges : privateIPv6AddressRanges;
```

Comment:
Minor performance improvement - can we sort the list so we can skip until we find a possible range for a given address? Since I assume this function returns false for most cases, will be good to find a way to fail fast - like find the absolute max of the priate address and return if the supplied address is greater than that.